### PR TITLE
🔀 Added Administrator Management based on registration tokens

### LINKED ISSUE #103: ✨ Create new registertokens to allow new admins
> **As a** User
> **I want** add other admins
> **so that** because either I want to access the bot on my smartphone as well or I sometimes I am on vacation and someone else should manage the bot

## Description:
After #102 has disabled the endpoints to add further admins. This user story is used to re implement this feature to add additional admins. As before the table should display the a list registered admin, and deletable admins can be detelted. Instead of an additional second panel, the "Add admin" button should be part of the the table header. If a admin requests a new registration token, a modal pops up.This modal should include: The token, the token in combination with the url as link and this link additional as QR Code.

### 🟢 In scope:
* Update Endpoints
* Update API-Documentation
* Update Frontend
* Update buefy to newest version
* Extend the token part of start up process to include the token as part of the link
* 
## What should be the result?
A user ...

* should be able to create new register tokens
* should be able to remove deletable admins

### PR DESCRIPTION
**Pull request by tjarbo**
> **As a** User
> **I want to** add other admins
> **because** either I want to access the bot on my smartphone as well or I sometimes I am on vacation and someone else should manage the bot 

Closes #103

## Description
This PR allows administrators to request further registration tokens. These tokens are used to let other people register themself as administrator. Administrators that are registered with these additional tokens can be removed again.

### Screenshots
image
image

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Checklist

- [ ] Make sure you are requesting to main. Don't request to release branch!
- [x]I have linked the related issue.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/backend/src/controllers/administrator/authenticator.schema.ts [hunk 1]
```diff
@@ -10,12 +10,10 @@ export interface IAuthenticatorDocument extends Document {
   transports?: AuthenticatorTransport[];
 }
 
-const authenticatorSchema = new Schema({
+export const authenticatorSchema = new Schema({
   credentialID: { type: Buffer, required: true },
   credentialPublicKey: { type: Buffer, required: true },
   counter: { type: Number, required: true },
   // ['usb' | 'ble' | 'nfc' | 'internal']
```
Comment (p-fruck) [inline packages/backend/src/controllers/administrator/authenticator.schema.ts:17]:
```suggestion
```

File: packages/backend/tests/administrator.spec.ts [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+    expect(mockNext.mock.calls.length).toBe(1);
+    expect(mockNext.mock.calls[0][0]).toEqual(new ApiError(403, `Administrator ${mockRequest.params.username} is not deletable`));
+
+  });
+
+  it.skip('should return 204 if deletion was successful', async () => {
+
+    /**
+     * Need to skip, because mocking deleteOne does not possible.
+     * Test failed due to timeout :()
```
Comment (p-fruck) [inline packages/backend/tests/administrator.spec.ts:200]:
```suggestion
     * Test failed due to timeout :(
```

File: packages/frontend/src/components/administration/RegistrationTokenModal.vue [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+            <div class="media-left">
+             <b-icon icon="qrcode" size="is-medium"></b-icon>
+            </div>
+            <div class="media-content">
+              <div class="content">
+                <p>
+                  <strong>QR-Code anzeigen</strong>
+                  <br />
+                  Klicke hier, um den Link als QR-Code anzuzeigen. Somit kannst du einfacher den
+                  Token auf, beispielweise, deinem Handy benutzten.
```
Comment (p-fruck) [inline packages/frontend/src/components/administration/RegistrationTokenModal.vue:52]:
```suggestion
                  Klicke hier, um den Link als QR-Code anzuzeigen. Somit kannst du beispielweise den
                  Token einfacher auf deinem Handy benutzten.
```

File: packages/frontend/src/views/Registration.vue [hunk 4]
```diff
@@ -67,6 +68,16 @@ export default {
       token: '',
     },
   }),
+  mounted() {
+    // Check is token paramter is available
```
Comment (p-fruck) [inline packages/frontend/src/views/Registration.vue:72]:
```suggestion
    // Check if token parameter is available
```

**Review by p-fruck — APPROVED** [review]

File: packages/frontend/src/components/administration/RegistrationTokenModal.vue [hunk 5]
```diff
@@ -48,8 +48,8 @@
                 <p>
                   <strong>QR-Code anzeigen</strong>
                   <br />
-                  Klicke hier, um den Link als QR-Code anzuzeigen. Somit kannst du einfacher den
-                  Token auf, beispielweise, deinem Handy benutzten.
+                  Klicke hier, um den Link als QR-Code anzuzeigen. Somit kannst du beispielweise
+                  den Token einfacher auf deinem Handy benutzten.
```
Comment (p-fruck) [inline packages/frontend/src/components/administration/RegistrationTokenModal.vue:52]:
```suggestion
                  den Token einfacher auf deinem Handy benutzen.
```
Missed that one :confused:

**Review by p-fruck — CHANGES_REQUESTED** [review]

**Review by p-fruck — APPROVED** [review]
:+1:

**Review by antonplagemann — CHANGES_REQUESTED** [review]
Nice 💯

**Review by antonplagemann — APPROVED** [review]
👍

**Pull request merged**